cmd/kubemark/hollow-node.go

```diff
@@ -131,6 +131,12 @@ func (c *hollowNodeConfig) createHollowKubeletOptions() *kubemark.HollowKubletOp
 }
 
 func main() {
+	if err := runHollowNodeCmd(); err != nil {
+		os.Exit(1)
```

Comment:
I think it is better to print error here instead of klog.Errorf on each line on `run()`.

Maybe, in the current implementation, we cannot print error from cobra, like flag parsing error.

Reply:
I think your pr for kubelet is fine on this because it tries to parse flag in RunE (and print error if flags are invalid).

https://github.com/kubernetes/kubernetes/pull/104774/files#diff-3a14a9838132c6668f4d100c26afa05b471b149c469edfece9b84fcd058c8a4bR157

Reply:
sorry, I'm a litter confused, since I returned every possible errors in `run()`, why would I miss some errors? Or you mean since I set `SilenceErrors=true`, some errors used to be print by cobra now are in silent yet.🧐

Reply:
sorry, It wasn't a good explanation. 🙏 

>  you mean since I set SilenceErrors=true, some errors used to be print by cobra now are in silent yet.🧐

Yes!

By SilenceErrors true, cobra don't print error( but it returns error).

If a user passes invalid flag, cobra return error, like `unknown flag: --invalid-flag`. 
Sorry I didn't try it actually, but I guess it exit(1) but not print any error with current implementation.

Reply:
cobra parses flags [here](https://github.com/spf13/cobra/blob/master/command.go#L946-L950)
and when it fail, cobra only returns err if we set selence error true ([ref](https://github.com/spf13/cobra/blob/master/command.go#L960))

Reply:
no `exit(1)` exists in `command.Execute()`, they are all returned as error or nil. I think it's not a matter.

Reply:
sorry, I cannot catch what you mean.

---

I tried it. And it worked as I guess.
It doesn't print flag error.

```
 (*'-') <  ./_output/bin/kubemark --invalid-flag
Usage:
  kubemark [flags]

Flags:
      --add-dir-header                            If true, adds the file directory to the header of the log messages
      --alsologtostderr                           log to standard error as well as files
      --api-server-port int                       Port on which API server is listening. (default 443)
      --azure-container-registry-config string    Path to the file containing Azure container registry configuration information.
      --boot-id-file string                       Comma-separated list of files to check for boot-id. Use the first one that 

.... (super long usage message)
```

We want to print flag parse error like below.
we can catch `unknown flag: --invalid-flag` error from returned error.

```
(*'-') < ./_output/bin/kube-proxy  --invalid-flag
Error: unknown flag: --invalid-flag
Usage:
  kube-proxy [flags]

Flags:
      --add-dir-header                               If true, adds the file directory to the header of the log messages
      --alsologtostderr                              log to standard error as well as files
      --bind-address ip                              The IP address for the proxy server to serve on (set to '0.0.0.0' for all IPv4 interfaces and '::' for all IPv6 interfaces) (default 0.0.0.0)

.... (super long usage message)
```

Reply:
thanks a lot for your help. I think the `Error: unknown flag` is located [here](https://github.com/spf13/cobra/blob/de187e874d1ca382320088f8f6d76333408e5c2e/command.go#L783-L786)

whatever, finally I disable the `SilenceErrors` and print the errors by cobra itself.  See [here](https://github.com/spf13/cobra/blob/de187e874d1ca382320088f8f6d76333408e5c2e/command.go#L983-L993), I think it solved the problem. Thanks again for your patience.👍

Reply:
> disable the SilenceErrors and print the errors by cobra itself.

Yes, it is also fine. thanks :)